### PR TITLE
Documentation for GetContractTypeId and GetParties in both scaladsl and javadsl

### LINKED ISSUE #15852: Documentation for GetContractTypeId and GetParties in both scaladsl and javadsl
From https://github.com/digital-asset/custom-views/pull/13

### PR DESCRIPTION
**Pull request by chunlokling-da**

### CODE REVIEW

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 1]
```diff
@@ -135,13 +135,24 @@ object BatchSource {
...
+
+    /** Extracts an optional `Identifier` from `E`. */
     def from(event: E): ju.Optional[J.Identifier]
+
+    /** Converts to scala DSL types. */
     def toScala: SGetContractTypeId[E] = (event: E) =>
       from(event).toScala.map(i => Identifier.fromJavaProto(i.toProto))
   }
 
+  /** Methods that instantiates [[GetContractTypeId]] from common event types */
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]:
```suggestion
  /** Provides methods that create [[GetContractTypeId]]s from common event types */
```

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 2]
```diff
@@ -135,13 +135,24 @@ object BatchSource {
     }
   }.toJava
 
+  /**
+   * Returns an optional `Identifier` from `E`.
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]:
```suggestion
   * Extracts an optional `Identifier` from an `E` element. It is used in [[BatchSource]] `create` methods used for testing purposes, where the [[BatchSource]] is created from in-memory data structures.
```

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 3]
```diff
@@ -135,13 +135,24 @@ object BatchSource {
...
+   *
+   * It should be only used by [[BatchSource]] create methods which creates [[BatchSource]] for testing purposes. It can
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]:
```suggestion
```

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 4]
```diff
@@ -135,13 +135,24 @@ object BatchSource {
...
+   * be instantiated with `from` methods in object [[GetContractTypeId]].
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]: The [[GetContractTypeId] object provides `from` methods for common events.

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 5]
```diff
@@ -162,12 +173,23 @@ object BatchSource {
       (exercisedEvent: J.ExercisedEvent) => Optional.of(exercisedEvent.getTemplateId)
   }
 
+  /**
+   * Returns an parties from `E`.
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]: Extracts  parties from `E`.

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 6]
```diff
@@ -162,12 +173,23 @@ object BatchSource {
...
+   *
+   * It should be only used by [[BatchSource]] create methods which creates [[BatchSource]] for testing purposes. It can
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala:179]: It is used in [[BatchSource]] `create` methods used for testing purposes, where the [[BatchSource]] is created from in-memory data structures.

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 7]
```diff
@@ -162,12 +173,23 @@ object BatchSource {
...
+   * be instantiated with `from` methods in object [[GetParties]]
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]:
```suggestion
The [[GetParties]] object provides `from` methods for common events.
```

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala (hunk 6 above)
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]:
```suggestion
* It is used in [[BatchSource]] `create` methods used for testing purposes, where the [[BatchSource]] is created from in-memory data structures.```

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 8]
```diff
@@ -162,12 +173,23 @@ object BatchSource {
...
+   */
   @FunctionalInterface
   trait GetParties[E] {
+
+    /** Extracts an set of parties from `E`. */
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]:
```suggestion
    /** Extracts a set of parties from `E`. */
```

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 9]
```diff
@@ -162,12 +173,23 @@ object BatchSource {
...
     def from(event: E): ju.Set[String]
+
+    /** Converts to scala DSL types. */
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]:
```suggestion
```

Comment (ray-roestenburg-da) [reply src/main/scala/com/daml/projection/javadsl/BatchSource.scala]: We have not provided docs for toJava and toScala so far.

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 10]
```diff
@@ -162,12 +173,23 @@ object BatchSource {
...
     def toScala: SGetParties[E] = (event: E) => from(event).asScala.toSet
   }
 
+  /** Methods that instantiates [[GetParties]] from common event types */
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]:
```suggestion
Provides methods that create [[GetParties]] from common event types */
```

File: src/main/scala/com/daml/projection/scaladsl/BatchSource.scala [hunk 11]
```diff
@@ -206,13 +206,23 @@ object BatchSource {
     }
   }
 
+  /**
+   * Returns an optional `Identifier` from `E`.
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/scaladsl/BatchSource.scala]: Please provide similar text as in the suggestions in the javadsl

File: src/main/scala/com/daml/projection/scaladsl/BatchSource.scala [hunk 12]
```diff
@@ -232,13 +242,23 @@ object BatchSource {
       (exercisedEvent: ExercisedEvent) => exercisedEvent.templateId
   }
 
+  /**
+   * Returns an parties from `E`.
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/scaladsl/BatchSource.scala]: Same here, use suggestions like in the java counterpart

**Review by ray-roestenburg-da — COMMENTED** [review]
Some suggestions to process, please do the same on the scaladsl, which I have not created explicit suggestions for.

**Comment by chunlokling-da** [comment]
Thanks @ray-roestenburg-da comment addressed

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 13]
```diff
@@ -135,13 +135,23 @@ object BatchSource {
     }
   }.toJava
 
+  /**
+   * Extracts an optional `Identifier` from an `E` element. It is used in [[BatchSource]] `create` methods used for
+   * testing purposes, where the [[BatchSource]] is created from in-memory data structures.
+   *
+   * The [[GetContractTypeId]] object provides from methods for common events.
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]:
```suggestion
   * The [[GetContractTypeId]] object provides `from` methods for common events.
```

File: src/main/scala/com/daml/projection/scaladsl/BatchSource.scala [hunk 14]
```diff
@@ -206,13 +207,22 @@ object BatchSource {
     }
   }
 
+  /**
+   * Extracts an optional `Identifier` from an `E` element. It is used in [[BatchSource]] `create` methods used for
+   * testing purposes, where the [[BatchSource]] is created from in-memory data structures.
+   *
+   * The [[GetContractTypeId]] object provides from methods for common events.
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/scaladsl/BatchSource.scala]:
```suggestion
   * The [[GetContractTypeId]] object provides `from` methods for common events.
```

File: src/main/scala/com/daml/projection/javadsl/BatchSource.scala [hunk 15]
```diff
@@ -162,12 +172,22 @@ object BatchSource {
       (exercisedEvent: J.ExercisedEvent) => Optional.of(exercisedEvent.getTemplateId)
   }
 
+  /**
+   * Extracts parties from `E` element. It is used in [[BatchSource]] `create` methods used for testing purposes, where
+   * the [[BatchSource]] is created from in-memory data structures.
+   *
+   * The [[GetParties]] object provides from methods for common events.
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/javadsl/BatchSource.scala]:
```suggestion
   * The [[GetParties]] object provides `from` methods for common events.
```

File: src/main/scala/com/daml/projection/scaladsl/BatchSource.scala [hunk 16]
```diff
@@ -232,13 +242,22 @@ object BatchSource {
       (exercisedEvent: ExercisedEvent) => exercisedEvent.templateId
   }
 
+  /**
+   * Extracts parties from `E` element. It is used in [[BatchSource]] `create` methods used for testing purposes, where
+   * the [[BatchSource]] is created from in-memory data structures.
+   *
+   * The [[GetParties]] object provides from methods for common events.
```
Comment (ray-roestenburg-da) [inline src/main/scala/com/daml/projection/scaladsl/BatchSource.scala]:
```suggestion
   * The [[GetParties]] object provides `from` methods for common events.
```

**Review by ray-roestenburg-da — APPROVED** [review]

**Pull request merged**